File: rabbits-in-forest/rabbits-in-forest.java
class Solution {
   public int numRabbits(int[] answers) {
		int[] freq = new int[1000]; //Size is given in contraints
		int ans = 0;
		for (int ele : answers) {
			if(freq[ele] == 0){
				ans+=(ele + 1); 
			}
			freq[ele]++;
			if(freq[ele] == ele + 1) freq[ele] = 0;
		}
		return ans;
	}
}

File: 797-rabbits-in-forest/rabbits-in-forest.js
/**
 * @param {number[]} answers
 * @return {number}
 */
var numRabbits = function (answers) {
    const map = new Map();
    let total = 0;

    for (let ans of answers) {
        if (ans === 0) {
            total += 1;
        } else if (!map.has(ans)) {
            map.set(ans, 1);
            total += ans + 1;
        } else {
            map.set(ans, map.get(ans) + 1);
            if (map.get(ans) > ans + 1) {
                map.set(ans, 1);
                total += ans + 1;
            }
        }
    }
    return total;
};